pkg/internal/ebpf/common/spanner.go

```diff
@@ -95,3 +101,45 @@ func extractTraceparent(traceparent [55]byte) string {
 	}
 	return string(traceparent[:])
 }
+
+func getSQLOperationAndTable(queryString string) (string, string) {
+	fields := strings.Fields(queryString)
+	if len(fields) == 0 {
+		return "", ""
+	}
+	operation := strings.ToUpper(fields[0])
+	if len(fields) < 2 {
+		return operation, ""
+	}
+	table := ""
+	if operation == "UPDATE" {
+		return operation, fields[1]
+	}
+	if len(fields) < 3 {
+		return operation, ""
+	}
+	switch operation {
```

Comment:
This needs to be a bit more complex, we need to skip tokens until we find one of the ones we are looking for. For example, 

WITH, UNION

Reply:
Honestly, looking at how brittle this parsing might be and all the edge cases, did you perhaps consider in your design using a library to parse the SQL statement? Was there a performance or dependency reason why not to use something like https://github.com/xwb1989/sqlparser?

Reply:
That parser was my first plan, but I was told that performance-wise, we do not want to do a full parse on each operation, that we should try to do something simple to get the operation and table that works in most (but not necessarily all) cases.  I don't think we're going to end up with something good if we make our own parser that goes beyond the simplest case such as only finding the first word and assuming that it is actually something useful, such as the db operation.  Perhaps we may need to end up using a real parser.  The config option that controls copying the DB statement could disable it if the performance is unacceptable.  We'd still get the DB spans/metrics, just without the operation/table name.  One downside would be if the db query statement is truncated due to limited buffer size in ringbuf and the parse fails, it may just waste a lot of time.
What do you think, is it likely that using xwb1989 would create too much of a performance impact?  Would it be overly complicated to have a tri-state config: 1) use full sql parser, 2) use first word only, 3) don't process db statement at all.

Reply:
Yeah, that makes sense about the performance, thanks! Let me answer in order what my thoughts are: 

> The config option that controls copying the DB statement could disable it if the performance is unacceptable. We'd still get the DB spans/metrics, just without the operation/table name.

I don't think this is an issue, we are doing application observability with Beyla and the overhead is expected to be higher, copying buffers overhead should be minimal here. We already do that for HTTP/gRPC. In a sense, number of queries without additional details can be determined with Infrastructure O11y, where App O11y does better/more is getting the level of detail that helps the developer.

> What do you think, is it likely that using xwb1989 would create too much of a performance impact? 

I think the only way to find out is to make a small benchmark to see how much slower is it on some sample queries, but I think the other point you made is stronger against this suggestion. If the parser fails completely on an incomplete query, without returning anything, then it's probably not important if it's fast enough. Does the parser fail without returning even a table name and operation if the query string is truncated?

If the parser is not an option, I think the code can be changed to loop a bit until it finds what it wants. Given that in 99% of cases it will be only one loop iteration, it shouldn't matter for performance. I think TOP would be common in MSSQL, but not elsewhere. But skipping WITH and RECURSIVE and other keywords until we find what the correct op is should be done. It's maybe worthwhile looking at docs in few SQL engines to see all the traps and building something more robust.

Reply:
I'm working on a new approach: Use xwb1989 tokenizer Scan() to find the type of each next token.  Find DB operation by matching against the set of db operation types, and find the (presumed) table name finding the value of the first "ID" type token that it finds.
I think this should be a reasonable balance: we don't fully parse it (so we don't end up failing when something is truncated,) we don't directly handle the tokens and all the nuance ourselves, we can reliable determine if something is a operation or a name, however it may be a wrong assumption that the first found "ID" type token would be a table name, although that should not be a sensitive token, so I think the risk is very low of leaking sensitive data values.